### PR TITLE
Move OAuth parameters outside public context

### PR DESCRIPTION
**Pull request by aeris**

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -641,6 +648,7 @@ func UseViper(v *viper.Viper) error {
 		},
 		Contexts:   v.GetStringMap("contexts"),
 		Registries: regs,
+		Clouderies: clouderies,
```
Comment (nono) [inline pkg/config/config.go]: Why you don't use `v.GetStringMap("clouderies")` like for the contexts?

File: pkg/config/config.go [hunk 2]
```diff
@@ -439,6 +440,12 @@ func UseViper(v *viper.Viper) error {
 		return err
 	}
 
+	tmp := v.Get("clouderies")
```
Comment (nono) [inline pkg/config/config.go]: Can you also update the example config file? https://github.com/cozy/cozy-stack/blob/master/cozy.example.yaml

File: pkg/instance/manager.go [hunk 3]
```diff
@@ -8,7 +8,11 @@ import (
 	"strings"
 	"time"
 
-	"github.com/cozy/cozy-stack/pkg/utils"
+	"github.com/cozy/cozy-stack/pkg/config"
```
Comment (nono) [inline pkg/instance/manager.go]: Do you use `gofmt` or `goimports`? It is weird to have a blank line here

Comment (aeris) [reply pkg/instance/manager.go]: I use goimports

Comment (nono) [reply pkg/instance/manager.go]: It's weird, goimports remove the blank lines on my machine.

File: pkg/instance/manager.go [hunk 4]
```diff
@@ -26,35 +30,89 @@ const (
...
-// ManagerURL returns an external string for the given ManagerURL kind.
-func (i *Instance) ManagerURL(k ManagerURLKind) (s string, ok bool) {
-	defer func() {
-		if !ok {
-			s = i.PageURL("/manager_url_is_not_specified", nil)
-		}
-	}()
+type managerConfig struct {
+	Api *struct {
+		Url   string
```
Comment (nono) [inline pkg/instance/manager.go]: Can you use `API` and `URL` like it's the convention in go? Cf https://github.com/golang/go/wiki/CodeReviewComments#initialisms

File: pkg/instance/manager.go [hunk 5]
```diff
@@ -26,35 +30,89 @@ const (
...
+	if err != nil {
+		return "", nil
+	}
+
+	base, ok := config["manager_url"]
+	if !ok {
+		return "", nil
+	}
+
+	baseUrl, err := url.Parse(base.(string))
```
Comment (nono) [inline pkg/instance/manager.go]: Same here, with `s/baseUrl/baseURL/`

File: pkg/instance/manager.go (hunk 4 above)
Comment (aeris) [reply pkg/instance/manager.go]: Done

File: pkg/config/config.go (hunk 2 above)
Comment (aeris) [reply pkg/config/config.go]: Done

File: pkg/instance/manager.go (hunk 5 above)
Comment (aeris) [reply pkg/instance/manager.go]: Done

**Pull request merged**